File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include <string>
#include <iostream>
#include <stack>

using namespace std;

bool solution(string s)
{
    bool answer = true;
    stack<char> S;
    
    for(int i=0; i<s.size(); i++){
        if(s[i]=='(') S.push(s[i]);
        else {
            if(!S.empty()) S.pop();
            else S.push(')');
        }
    }
    
    if(!S.empty()) return false;

    return answer;
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    stack=[]
    
    for i in s:
        if i=='(':
            stack.append(i)
        else :
            if len(stack)==0 :
                return False
            else :
                stack.pop()
    
    if len(stack)!=0 :
        answer=False
        
    return answer